bitcoin/src/blockdata/transaction/extended.rs

```diff
@@ -0,0 +1,133 @@
+    pub fn into_transaction(self) -> Transaction {
+        Transaction {
+            version: self.version,
+            input: self.inputs.into_iter().map(|input| input.txin).collect(),
+            output: self.outputs,
+            lock_time: self.lock_time,
+        }
+    }
+
+    /// Creates non-extended transaction from extended transaction.
+    ///
+    /// As opposed to `into_transaction` this allocates for each input and output however, as
+    /// unlike `tx.clone().into_transaction()` this avoids cloning the extra input data.
+    pub fn to_transaction(&self) -> Transaction {
+        Transaction {
+            version: self.version,
+            input: self.inputs.iter().map(|input| input.txin.clone()).collect(),
+            output: self.outputs.clone(),
+            lock_time: self.lock_time,
+        }
+    }
+
+    /// Computes the [`Txid`].
+    ///
+    /// Hashes the transaction **excluding** the segwit data (i.e. the marker, flag bytes, and the
+    /// witness fields themselves). For non-segwit transactions which do not have any segwit data,
+    /// this will be equal to [`Transaction::wtxid()`].
+    pub fn txid(&self) -> Txid {
+        use crate::consensus::Encodable;
+        use crate::hashes::Hash;
+
+        let mut enc = Txid::engine();
+        self.version.consensus_encode(&mut enc).expect("engines don't error");
+        VarInt(self.inputs.len() as u64).consensus_encode(&mut enc).expect("engines don't error");
+        for input in &self.inputs {
+            input.txin.consensus_encode(&mut enc).expect("engines don't error");
+        }
+        self.outputs.consensus_encode(&mut enc).expect("engines don't error");
+        self.lock_time.consensus_encode(&mut enc).expect("engines don't error");
+        Txid::from_engine(enc)
+    }
+
+    /// Returns the fee this transaction is paying.
+    ///
+    /// Returns `None` in case of overflow (creating bitcoins out of thin air).
+    pub fn fee(&self) -> Option<Amount> {
+        let in_amount = self.inputs.iter().map(|input| input.previous_output.value).checked_sum()?;
+        let out_amount = self.outputs.iter().map(|output| output.value).checked_sum()?;
+        in_amount.checked_sub(out_amount)
+    }
+
+    /// Returns the weight of this transaction, as defined by BIP-141.
+    pub fn weight(&self) -> Weight {
+        let inputs = self.inputs
+            .iter()
+            .map(|input| {
+                let witness_elements = input.txin.witness.iter().map(|element| element.len());
+                InputWeightPrediction::new(input.txin.script_sig.len(), witness_elements)
+            });
+        let outputs = self.outputs.iter().map(|output| output.script_pubkey.len());
+        super::predict_weight(inputs, outputs)
+    }
+
+    /// Returns the fee rate of this transaction.
+    ///
+    /// Returns `None` in case of overflow (creating bitcoins out of thin air).
+    pub fn fee_rate(&self) -> Option<FeeRate> {
```

Comment:
what about `long_term_fee_rate`?

Reply:
What?

Reply:
In core, spending a txout is more wasteful if the fee is greater than the long_term_fee (IE it's expensive to spend now).

https://github.com/bitcoin/bitcoin/blob/master/src/wallet/coinselection.cpp#L165

So, to do this computation, the fee and long_term_fee for a UTXO needs to be known.

Reply:
https://github.com/bitcoin/bitcoin/blob/03752444cd54df05a731557968d5a9f33a55c55c/src/wallet/coinselection.cpp#L165

Reply:
https://github.com/bitcoin/bitcoin/blob/03752444cd54df05a731557968d5a9f33a55c55c/src/wallet/coinselection.cpp#L165

Reply:
Is there some longer explanation of it? How is it computed?